src/redempt/redlib/region/SpheroidRegion.java

```diff
@@ -383,7 +385,61 @@ public boolean surfaceContains(Block block) {
 		}
 		return surface.contains(block);
 	}
-	
+
+	/**
+	 * Gets a random block within a spherical region
+	 * @return Bukkit block object of the randomly picked block
+	 */
+	public Block getRandomBlock() {
+		int[] randomBlockLocation = getRandomBlockCoordinates();
+		Location pickedLocation = new Location(getWorld(), randomBlockLocation[0], randomBlockLocation[1], randomBlockLocation[2]);
+		while (!contains(pickedLocation)) {
+			pickedLocation = new Location(getWorld(), randomBlockLocation[0], randomBlockLocation[1], randomBlockLocation[2]);
+		}
+		return pickedLocation.getBlock();
+	}
+
+	/**
+	 * Gets random block within spherical region
+	 * @param random Random instance
+	 * @return Bukkit block object of the randomly picked block
+	 */
+	public Block getRandomBlock(Random random) {
+		int[] randomBlockLocation = getRandomBlockCoordinates(random);
+		Location pickedLocation = new Location(getWorld(), randomBlockLocation[0], randomBlockLocation[1], randomBlockLocation[2]);
+		while (!contains(pickedLocation)) {
```

Comment:
This is never going to change, since the random block coordinates are never re-randomized.